### ISSUE TITLE
go to line should select text from the current cursor when holding down shift

### ISSUE THREAD

**Issue by tieTYT** [microsoft/vscode#97473]
This would be a very useful feature for those using dictation software to program. Because I am, when I want to highlight a few lines of text, I use "go to line" to get my start destination. It would be nice if I could hold down shift and then use "go to line" again and have it highlight the lines from my current cursor position to the 2nd line I specified (whether or not the 2nd line is above or below the current one).

Because holding shift doesn't work this way, I have to use dictation software to say "move up/down x lines." It's not a very intuitive way to think. I almost always over or under shoot the number of lines I need and have to readjust, whereas somebody with a keyboard wouldn't mind that because they have much faster feedback.

If there is a workaround to this for an extension that adds this functionality, I would be very interested in it. Thanks

**Comment by alexdima** [comment]
@isidorn I think this request is very similar to what we discussed in https://github.com/microsoft/vscode/issues/95894 with "Mark Start of Selection", "Mark End of Selection", "Go to Selection Mark".

@tieTYT I think if we had such commands, you could create a selection mark (anchor) and then you can move the cursor around with any command (including go to line), then execute the select from anchor command.

@isidorn Now that I had to use the names you have proposed, I have to say I'm not loving them, because if you select from bottom to top, then the start of the selection would be after the end of the selection with the proposed names.

**Comment by isidorn** [comment]
This is a fair feature request, however let's first add  #95894 and see if this is still needed
@alexdima thanks for feedback on the naming, agree it is not nice when the start is after the end. Will re-think it tomorrow when I plan to add it.

**Comment by tieTYT** [comment]
If #95894 were added, I really wouldn't need this feature very much. That said, I think this feature is a very intuitive way of working.

**Comment by isidorn** [comment]
We plan to add that feature some time this week. Thus closing this one.
In case you want to try that one out, you can try using VS Code insiders, since it will have the feature as soon as we add it.

**Comment by tieTYT** [comment]
@isidorn Are you saying a week from now it will be on insiders for a week from now it will be generally available?

**Comment by isidorn** [comment]
@tieTYT week from now in insiders. 3 weeks from now generally available
